File: backend/KanbanApi/Models/Ticket.cs
using System.ComponentModel.DataAnnotations;

namespace KanbanApi.Models;

public class Ticket
{
    public int Id { get; set; }

    [Required]
    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public decimal TimeSpentHours { get; set; }

    public DateTime CreatedAt { get; set; }

    public int Position { get; set; }

    public int KanbanColumnId { get; set; }
    public KanbanColumn? KanbanColumn { get; set; }
}

File: frontend/kanban-frontend/src/app/core/models/ticket.ts
export interface Ticket {
  id: number;
  title: string;
  description: string;
  timeSpentHours: number;
  createdAt: string;
  position: number;
  kanbanColumnId: number;
}

export interface CreateTicketDto {
  title: string;
  description: string;
  timeSpentHours: number;
  position: number;
  kanbanColumnId: number;
}

export interface UpdateTicketDto {
  title: string;
  description: string;
  timeSpentHours: number;
  position: number;
  kanbanColumnId: number;
}
